overlord/snapstate/snapstate_install_test.go

```diff
@@ -4180,3 +4180,38 @@ func (s *validationSetsSuite) TestInstallSnapInvalidByValidationSetIgnoreValidat
 	}
 	c.Assert(s.fakeBackend.ops[1], DeepEquals, expectedOp)
 }
+
+func (s *snapmgrTestSuite) TestInstallPrerequisiteWithSameDeviceContext(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	// this store should receive no download requests
```

Comment:
Nitpick, maybe `// unset the global store, it will need to come via the device context`  (if found the comment from existing tests a bit more clear).